fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/shim/ChaincodeBase.java

```diff
@@ -115,13 +131,13 @@
 
     public void start(final String[] args) {
         try {
+            initializeLogging();
```

Comment:
just to check - if there's an interaction between logging and the property controlled logging?

Reply:
i.e. if the logging is setup here - can the default log level be changed via environment variable etc.

Reply:
As far as I could tell, initializeLogging() processes its own env vars, so it should be ok...

https://github.com/hyperledger/fabric-chaincode-java/blob/master/fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/shim/ChaincodeBase.java#L171
https://github.com/hyperledger/fabric-chaincode-java/blob/master/fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/shim/ChaincodeBase.java#L184

If not, it would need fixing in ContractRouter which already initialises logging earlier. (I really wanted to do more tidy up between the two but I resisted and just put logging at the same position in the start up.)

https://github.com/hyperledger/fabric-chaincode-java/blob/master/fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/contract/ContractRouter.java#L55